libopenage/gamestate/game_main.cpp

```diff
@@ -37,17 +38,14 @@ GameMain::GameMain(const Generator &generator)
 }
 
 GameMain::~GameMain() {
-	for (auto &p : this->players) {
-		delete p;
-	}
 }
 
 unsigned int GameMain::player_count() const {
 	return this->players.size();
 }
 
 Player *GameMain::get_player(unsigned int player_id) {
-	return this->players.at(player_id);
+	return this->players.at(player_id).get();
```

Comment:
- How it does not break anything...? All `Player*` must become `std::shared_ptr<Player>`, returning a shared_ptr and calling get() right after has no point.
- "not comfortable enough" = 15% I don't know if I should + 15% I don't know if I can + 70% I don't want to
But if you incιst I can try